doc/testing_snitch.md

```diff
@@ -0,0 +1,146 @@
+# Testing _snitch_
+
+<!-- MarkdownTOC autolink="true" -->
+
+- [Why](#why)
+- [The _snitch_ test suite](#the-snitch-test-suite)
+    - [Run-time unit tests](#run-time-unit-tests)
+    - [Compile-time unit tests](#compile-time-unit-tests)
+    - [Approval tests](#approval-tests)
+        - [How to update approval test data](#how-to-update-approval-test-data)
+    - [Install tests](#install-tests)
+- [How to run the tests](#how-to-run-the-tests)
+    - [Running tests locally](#running-tests-locally)
+        - [Run-time unit tests](#run-time-unit-tests-1)
+        - [Compile-time unit tests](#compile-time-unit-tests-1)
+        - [Approval tests](#approval-tests-1)
+        - [Install tests](#install-tests-1)
+    - [Automated testing pipeline](#automated-testing-pipeline)
+
+<!-- /MarkdownTOC -->
+
+## Why
+
+If you are trying to [contribute](../CONTRIBUTING.md) to _snitch_, you should run the _snitch_ test suite to ensure your changes have not accidentally broken any existing feature. You should also ensure that any new feature you add is properly tested, so others don't break it with their changes later on.
```

Comment:
I think it's more robust to use an "absolute" path here - `/CONTRIBUTING.md`. The same goes for `../.github/workflows`. See https://github.com/snitch-org/snitch/commit/9caaeec74611b626b0901e9af3c16ee4e4aadb44.

Reply:
Yep, done